### DISCUSSION TITLE
Manual date input (without datepicker)

### DISCUSSION THREAD

**Discussion by madex1** [directus/directus#3728] [Feature Requests]
It's rather annoying to look for day and month in the list. It would be nice to add ability to enter date manually :)

**Comment by benhaynes** [comment]
I agree. There are a few different ways that we could/should allow managing dates. These could be different interfaces, or interface options... depending on how different they are.

* Calendar
* Dropdowns
* Multiple Inputs
* Single Input (system)

Also, any of the last three could have an icon/button that opens up the calendar chooser as an dropdown.

**Reply by rijkvanzanten** [reply]
A nice in-between version would be to allow you to type in values in the dropdowns for day/month/year 🤔 Our dropdowns technically support "input" on the "preview" bit as well, so that is a possibility

**Reply by madex1** [reply]
What about single input with mask depending on local date format? i.e. MM-DD-YYYY for US and MM.DD.YYYY for Europe?

**Reply by vanling** [reply]
Yes! this is what i came across today. I'm using the time field for project/work logging and would be nice if i could type "1" or "2:30" directly in the field.

**Comment by Michael-Nussbaumer** [comment]
Lets say the interface for date/datetime is not directus like right now. I am also looking for alternatives right now, https://flatpickr.js.org/ looks interesting for me.

**Reply by uebmaster** [reply]
I like the flatpick proposal, in fact i've arrived here making a search about the calendar display, cause i think that the display must to show the years inversely, actually looks like the image:

![image](https://user-images.githubusercontent.com/5565506/129622026-0c57bb3f-3713-4cd4-831a-b318140e645e.png)

But if i am asking the birthday and i see that the years advance when i scroll down, then by logic, if I need to choose a year back to 2016, then I would have to go up instead of down to find or write the year I want, for example 1995, actually i can't go back 2016 if i scroll up, I don't know if I can explain myself?

or failing that there should be an option to choose in which year you want the scroll to start.

I don't know if my suggest can be post it here or i need to open a new pr.

Thanks in advance.

**Reply by mark-james** [reply]
Hi @MichaelNussbaumerGOWEST,
Did you get anywhere with the flatpickrjs library as a directus interface?

Like you, I'm keen to leverage a calendar picker within directus. 

cheers
Mark

**Reply by benhaynes** [reply]
I should also mention that we are interested in switching to a calendar-based date interface. Anyone who is looking to help create this should reach out so we can coordinate an official core component. We'd love to help! ❤️

**Reply by Michael-Nussbaumer** [reply]
> Hi @MichaelNussbaumerGOWEST,
> Did you get anywhere with the flatpickrjs library as a directus interface?
> 
> Like you, I'm keen to leverage a calendar picker within directus.
> 
> cheers
> Mark

No I haven't found the time yet, as my priorities shifted the last few month to a non directus project.

**Reply by mark-james** [reply]
Thanks Michael.

Ben, happy to talk this one through. What do you have in mind for the approach?

Wrap flatpickrjs in a vue component?

Or something else?

**Reply by benhaynes** [reply]
As always, ideally, I would like to maintain some control over the UI/design of this component. That said, I know Rijk probably wants to use the browser/system input, since it now supports a calendar style.

Open to looking into any other existing libraries, but defer to @rijkvanzanten on the best technical path forward. ❤️

**Reply by rijkvanzanten** [reply]
The browser system one is the easiest and most reliable, but also oftentimes ugly and inflexible to integrate within the design system. That being said, I think the functionality of a calendar is so complex to build/maintain properly, that it's likely a good idea to go with the browser native one :

**Reply by mark-james** [reply]
No problem. That keeps things simple.
Does it make sense to update v-input to accept the types date and datetime?

On the interface side, do you want a new interface, or an update to the existing datetime one?

**Reply by benhaynes** [reply]
Regardless of what we decide stays in core, we should capture unique UI interfaces for the marketplace. Basically, this just means putting a copy on NPM.

**Reply by rijkvanzanten** [reply]
> Does it make sense to update v-input to accept the types date and datetime?

The v-input passes on the `type` prop to the underlying `<input>` element, so that should already work. I believe the input type we're looking for is `datetime-local` 

<img width="224" alt="CleanShot 2021-09-20 at 10 15 07@2x" src="https://user-images.githubusercontent.com/9141017/134017767-96d1dda9-ecba-4f26-b617-fd24832ec0c0.png">

**Reply by mark-james** [reply]
Hi @benhaynes, 
I've created a custom interface that works for the types date, time and dateTime. It uses v-input as the core component using the type prop as Rijk suggested. It's pretty simple, no user options at this stage.

How do you want to proceed? 

Happy to bundle it for the marketplace. Happy to attempt a pull request to make it a core interface.

**Reply by rijkvanzanten** [reply]
Instead of adding it as a separate interface, I'm wondering if we should simply replace the current datetime interface with this browser based approach! Thoughts @benhaynes?

**Reply by benhaynes** [reply]
Yes, I think we should use that for the system / core interface, potentially keeping the current drop-downs as an interface option... but we don't have to if that makes things overly complex.

**Reply by mark-james** [reply]
Ok, let me have a look at combining the two.

Regarding browser support, this native datepicker feature should be supported by all the browsers other than IE which apparently still has a few % of the market... It should degrade gracefully to a type=text. But I haven't tested this. What are your thoughts on handling this use case?

I guess we could switch to the current dropdowns for IE as well?

**Reply by rijkvanzanten** [reply]
> Regarding browser support, this native datepicker feature should be supported by all the browsers other than IE which apparently still has a few % of the market... It should degrade gracefully to a type=text. But I haven't tested this. What are your thoughts on handling this use case?
> I guess we could switch to the current dropdowns for IE as well?

Directus itself doesn't run at all on IE11, and we have no intention to add support either, so that should be totally fine!

**Reply by HemboPagi** [reply]
In our case we use dates as day-month-year format as well as just year. And we work with medieval dates, so we need to enter dates manually. In current date-picker interface i can not see the way just to pick year with out month and day.

**Comment by nordcart** [comment]
up

**Comment by joselcvarela** [comment]
Offtopic:

@nordcart to upvote an idea, you can click on this button. Please 🙏 do not make unnecessary comments.

![image](https://user-images.githubusercontent.com/14039341/134674811-29423972-d54f-422f-838b-5a0a2b0395e5.png)

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by MarkBurvs** [comment]
Did anything ever become of this? 
Our app relies heavily on date-inputting, and my users are consistently reporting that date-inputting is the single painful bottleneck in the process of entering data. They just want to be able to type dates.

**Comment by leonardoviada** [comment]
Any updates on this?

**Comment by rongshabing** [comment]
We would also be interested in this! Is this being worked on?

**Reply by JamesW1** [reply]
Hi There - This is not currently being worked on, you can track our current focus on the public roadmap here:
https://roadmap.directus.io/roadmap

If we get a chance to address this subject, I'll make sure to post an update on this thread.

Thanks,
James

**Reply by rongshabing** [reply]
Awesome, thanks a lot!

**Comment by azhirov** [comment]
It's very difficult to enter dates of birth

**Comment by MarkBurvs** [comment]
Yay... great to see this has been added in v12. 🎉

**Reply by JamesW1** [reply]
Thanks @MarkBurvs - for others interested, this was added in 12.0.1:

Video showing the behaviour on the PR - thanks @robluton!
https://github.com/directus/directus/pull/27693
